src/Raven.Server/Documents/Indexes/Workers/HandleReferences.cs

```diff
@@ -173,12 +180,11 @@ protected HandleReferencesBase(Index index, DocumentsStorage documentsStorage, I
                         {
                             UpdateReferences(indexContext, collection);
 
+                            RenewTransactionIfNeeded(queryContext, batchContinuationResult, indexed, ref readTransaction);
+
                             var hasChanges = false;
                             earlyExit = false;
 
-                            var indexed = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
-
-                            using (queryContext.OpenReadTransaction())
                             {
```

Comment:
This can be removed, but doing so causes a leftward shift, giving the impression of more changes than actually exist.
Once this is reviewed, I'll format it correctly.